glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -193,7 +129,6 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
      */
     override fun doWork(): Result {
         return when {
```

Comment:
nit: Using `when` here seems like overkill since there are only two paths now, maybe simplify this down to a one-liner like this:
```Kotlin
 return if (uploadPings()) Result.success() else Result.retry()
```

Reply:
Please don't do this as a one-liner.
Why don't we merge `uploadPings` into `doWork` and make it return the right result? (or just have `return uploadPings()` and make `uploadPings` return that result.

Reply:
(at the very least I find a multi-liner more readable:

```kotlin
if (uploadPings()) {
	return Result.success();
} else {
	return Result.retry();
}
```

but IMO never use a boolean to signal something if you have a more expressive type handy.

Reply:
The only reason I suggested a one-liner was because of the simplicity of the statement.  I agree that `uploadPings()` returning the `Result` value would probably be better.